cosmo_tester/test_suites/bootstrap_based_tests/sanity_scenario_test.py

```diff
@@ -0,0 +1,174 @@
+
+from cosmo_tester.test_suites.snapshots import (
+    create_snapshot,
+    download_snapshot,
+    upload_snapshot,
+    restore_snapshot,
+    # upgrade_agents,    # bug in agents upgrade
+    verify_services_status
+)
+
+# from cosmo_tester.test_suites.ha.ha_helper \      # bug in agents upgrade
+#     import HighAvailabilityHelper as ha_helper
+
+USER_NAME = "sanity_user"
+USER_PASS = "user123"
+TENANT_NAME = "tenant"
+
+
+@pytest.fixture(scope='module')
+def managers(cfy, ssh_key, module_tmpdir, attributes, logger):
+    """Bootstraps 3 Cloudify managers on a VM in Rackspace OpenStack."""
+
+    hosts = TestHosts(
+        cfy, ssh_key, module_tmpdir, attributes, logger,
+        number_of_instances=3)
+
+    for manager in hosts.instances[1:]:
+        manager.upload_plugins = False
+
+    try:
+        hosts.create()
+        yield hosts.instances
+    finally:
+        hosts.destroy()
+
+
+@pytest.fixture(scope='function')
+def nodecellar(request, cfy, managers, attributes, ssh_key, tmpdir, logger):
+    manager = managers[0]
+    manager.use()
+    tenant = prepare_and_get_test_tenant(TENANT_NAME, manager, cfy)
+    nc = NodeCellarExample(
+        cfy, manager, attributes, ssh_key, logger, tmpdir,
+        tenant=tenant, suffix='simple')
+    nc.blueprint_file = 'simple-blueprint-with-secrets.yaml'
+    yield nc
+
+
+def test_sanity_scenario(managers,
+                         cfy,
+                         logger,
+                         tmpdir,
+                         attributes,
+                         nodecellar):
+    manager1 = managers[0]
+    manager2 = managers[1]
+    manager3 = managers[2]
+
+    logger.info('Cfy version')
+    cfy('--version')
+
+    logger.info('Cfy status')
+    cfy.status()
+
+    # Create user and add the new user to the tenant
+    _configure_tenant(cfy, manager1, logger)
+
+    _set_sanity_user(cfy, manager1, logger)
+
+    # Creating secrets
+    _create_secrets(cfy, logger, manager1)
+
+    nodecellar.upload_and_verify_install()
+
+    _set_admin_user(cfy, manager1, logger)
+
+    _start_cluster(cfy, manager1, logger)
+
+    _set_admin_user(cfy, manager2, logger)
+
+    _join_cluster(cfy, manager1, manager2, logger)
+
+    logger.info('Setting replica manager')
+    # ha_helper.set_active(manager2, cfy, logger)   # bug in agents upgrade
+
+    snapshot_id = 'SNAPSHOT_ID'
+    local_snapshot_path = str(tmpdir / 'snap.zip')
+    logger.info('Creating snapshot')
+    create_snapshot(manager1, snapshot_id, attributes, logger)
+    download_snapshot(manager1, local_snapshot_path, snapshot_id, logger)
+
+    _set_admin_user(cfy, manager3, logger)
+
+    logger.info('Uploading and restoring snapshot')
+    upload_snapshot(manager3, local_snapshot_path, snapshot_id, logger)
+    restore_snapshot(manager3, snapshot_id, cfy, logger)
+    verify_services_status(manager3)
+    # wait a while to allow the restore-snapshot post-workflow commands to run
```

Comment:
This sleep is actually for agents reconnection

Reply:
fixed